fatf/transparency/models/feature_influence.py

```diff
@@ -623,3 +644,406 @@ def partial_dependence(dataset: np.ndarray,
+        predictive error will be alculated
+    model
+        A fitted model whose predictions will be used to calculate
+        associated predictive error based on 'scoring_metric'. (Please
+        see :class:`fatf.utils.models.models.Model` class documentation for the
+        expected model object specification.)
+    target: np.ndarray
+        A vector corresponding to truth values for class labels of ``dataset``
+    as_regressor: Optional[bool] = None
+        A bool variable used to signify that the ``model``
+        is a regression model. Used to inform the ``scoring_metric``
+    scoring_metric: str
+        The type of scoring measure used to evaluate the predictive perfomance
+        of ``model`` on ``dataset`` - is ``scoring_metric`` is None, the
+        function either applies scikit learn's default
+        scorer for that model if ``model`` is a scikit-learn model,
+        else applies accuracy for a classifier and max error for a regressor.
+    Returns
+    ---------------
+    score: float
+    Float representing the result of applying the ``scoring_metric``
+    (either default or user-inputted) performed on the outputs
+    of  the``model`` predictions on the ``dataset`` and
+    the ``target`` vector."""
+    if SKLEARN_MISSING is True:
+        if as_regressor:
+            predictions = model.predict(dataset)  # type: ignore
+            score = -np.max(np.abs(target - predictions))
+        else:
+            predictions = model.predict(dataset)  # type: ignore
+            confusion_matrix = fumt.get_confusion_matrix(target, predictions)
+            score = fumm.accuracy(confusion_matrix)
+    else:
+        if ftst.is_sklearn_model(model):
+            scorer = check_scoring(model, scoring_metric)
+        else:
+            if scoring_metric is None:
+                if as_regressor:
+                    scorer = check_scoring(model, 'max_error')
+                else:
+                    scorer = check_scoring(model, 'accuracy')
+            else:
+                scorer = check_scoring(model, scoring_metric)
+        score = scorer(model, dataset, target)
+    return score
+
+
+def permutation_feature_importance(dataset: np.ndarray,
+                                   model: object,
+                                   target: np.ndarray,
+                                   as_regressor: Optional[bool] = None,
+                                   scoring_metric: Optional[str] = None,
+                                   repeat_number: Optional[int] = None):
+    '''
+    Calculates the Permutation Feature Importance (PFI)
+    of each feature in a dataset.
+
+    Based on the provided ``dataset``, ``model``
+    and ground truth ``target`` vector, this function computes the
+    Permutation Feature Importance (PFI).
+    PFI works by
+    permuting the values of each feature and measuring
+    the change in prediction error compared to the original
```

Comment:
If we decide to change "prediction error" to "predictive performance metric" this needs to be changed here accordingly.